### PR TITLE
Codelive/rob cowell

### PR DESCRIPTION
**Pull request by codefriar**
### What does this PR do?
Introduces the 'Ouroboros' pattern of a self-re-enqueueing Queueable job for processing large amounts of data.

### CODE REVIEW

File: ruleset.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (msrivastav13) [inline ruleset.xml:1]: Can we get rid of this file or have specific rules than general categories? Please check this PR 

https://github.com/trailheadapps/apex-recipes/pull/146/files#diff-a6839fbfcf5024903ed122d22112e9e54011183ccc884da9d1e3a3c2a9d9e04c

**Review by msrivastav13 — CHANGES_REQUESTED** [review]

File: force-app/tests/LDV Recipes/LDVRecipes_Tests.cls [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        Test.startTest();
+        LDVRecipes ldvr = new LDVRecipes();
+        System.enqueueJob(ldvr);
+        Test.stopTest();
+
+        System.assertEquals(
+            1,
+            LDVRecipes.chunksExecuted,
+            'Expected to find the queueable executed 4 times'
```
Comment (msrivastav13) [inline force-app/tests/LDV Recipes/LDVRecipes_Tests.cls:33]: Pretty confusing messgae! Says 4 times executed while asserts 1.

File: force-app/main/default/classes/LDV Recipes/LDVRecipes.cls [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        String queryString = '';
+        queryString += 'SELECT ContentDocumentId,ContentDocument.Title, ContentDocument.CreatedDate,LinkedEntityId ';
+        queryString += 'FROM ContentDocumentLink ';
+        queryString += 'WHERE LinkedEntityId in (SELECT Id FROM Account) ';
+        if (offsetId != null) {
+            queryString += 'AND ID > :offsetId ';
+        }
+        queryString += 'ORDER BY ID ASC ';
+        queryString += 'LIMIT :chunkSize';
+        return Database.query(queryString);
```
Comment (msrivastav13) [inline force-app/main/default/classes/LDV Recipes/LDVRecipes.cls:86]: Surprised the `WITH SECURITY_ENFORCED` is skipped and PMD was happy with it!

**Review by msrivastav13 — APPROVED** [review]

File: force-app/main/default/classes/LDV Recipes/LDVRecipes.cls (hunk 3 above)
Comment (codefriar) [reply force-app/main/default/classes/LDV Recipes/LDVRecipes.cls:86]: You know, that's odd. I'm pushing 'WITH SECURITY_ENFORCED' to the branch

File: force-app/tests/LDV Recipes/LDVRecipes_Tests.cls (hunk 2 above)
Comment (codefriar) [reply force-app/tests/LDV Recipes/LDVRecipes_Tests.cls:33]: Good catch. This is an artifact of me trying to get re-entrant async tests working. (hint: they don't.)

**Pull request merged**